ee/clickhouse/queries/person_query.py

```diff
@@ -88,33 +104,18 @@ def _get_fields(self) -> List[Tuple[str, str]]:
         #   Here, we remove the ones only to be used for filtering.
         # The same property might be present for both querying and filtering, and hence the Counter.
         properties_to_query = self._column_optimizer._used_properties_with_type("person")
-        properties_to_query -= extract_tables_and_properties(self._filter.property_groups_flat)
-
-        if self._entity is not None:
-            properties_to_query -= extract_tables_and_properties(self._entity.properties)
+        if self._inner_person_properties:
+            properties_to_query -= extract_tables_and_properties(self._inner_person_properties.flat)
```

Comment:
This will now filter out select fields that are used on the inner query but not the outer